### PR TITLE
LPS-199854 Add an option to expire all versions in DL

### PR DESCRIPTION
**Pull request by robertoDiaz**

### CODE REVIEW

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -2596,6 +2599,28 @@ userId, _EMAIL_TYPE_EXPIRED, _buildEntryURL(latestFileVersion),
 		}
 	}
 
+	private void _expireFileVersion(
+			DLFileEntry fileEntry, DLFileVersion fileVersion, long userId,
+			Map<String, Serializable> workflowContext,
+			ServiceContext serviceContext)
```
Comment (marcogalluzzi) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2605]: I think parameter `serviceContext` should go before `workflowContext` (sorting reasons) and `userId` at the start

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]
Remember: as this change is a breaking change, the JIRA story should follow: https://liferay.atlassian.net/wiki/spaces/PD/pages/2387836929/Managing+Reporting+Product+Breaking+Changes#How-to-manage-Product-Breaking-Changes

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java (hunk 1 above)
Comment (robertoDiaz) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2605]: I use the same sort as used in existing calling method:

```
private void _expireFileEntriesByCompanyId(
			long companyId, Date expirationDate,
			Map<String, Serializable> workflowContext,
			ServiceContext serviceContext)
		throws PortalException {
```

Comment (robertoDiaz) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2605]: Why is a breaking change??

Default behavior is the same.

**Comment by robertoDiaz** [comment]
> Remember: as this change is a breaking change, the JIRA story should follow: https://liferay.atlassian.net/wiki/spaces/PD/pages/2387836929/Managing+Reporting+Product+Breaking+Changes#How-to-manage-Product-Breaking-Changes

Why is this a breaking change?

Default behavior is the same (expire only the last version)

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java (hunk 1 above)
Comment (robertoDiaz) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2605]: I could move the userId if you want, anyway

Comment (marcogalluzzi) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:2605]: OK, keep using the same order then. But at least I would move `userId` parameter.

**Comment by marcogalluzzi** [comment]
> https://liferay.atlassian.net/wiki/spaces/PD/pages/2387836929/Managing+Reporting+Product+Breaking+Changes#How-to-manage-Product-Breaking-Changes

The "Breaking change" is having a new option in the configuration that gives the user a new functionality. That's what I understand for _breaking change_ if I read the documentation.

https://liferay.atlassian.net/wiki/spaces/PD/pages/2387836929/Managing+Reporting+Product+Breaking+Changes#What-Breaking-Changes-are

**Comment by robertoDiaz** [comment]
I guess that you are refering to this:

> Configuration changes: Changes in default configurations or files like portal.properties, system.properties, OSGI configs, etc, that change the default behavior of the Portal. This excludes  the removal of Beta and Release feature flags (these changes are communicated separately).

But I do not " change default configurations or files like portal.properties, system.properties, OSGI configs, etc, that change the default behavior of the Portal". 

I'm adding a new option that do not changes this default behavior... 🤔

**Comment by marcogalluzzi** [comment]
> I guess that you are refering to this:
> 
> > Configuration changes: Changes in default configurations or files like portal.properties, system.properties, OSGI configs, etc, that change the default behavior of the Portal. This excludes  the removal of Beta and Release feature flags (these changes are communicated separately).
> 
> But I do not " change default configurations or files like portal.properties, system.properties, OSGI configs, etc, that change the default behavior of the Portal".
> 
> I'm adding a new option that does not change current default behavior... 🤔

We may need a conversation with product management. In regards to this PR just apply the requested change so we can move forward.

**Comment by robertoDiaz** [comment]
Sure! 👍

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java (hunk 1 above)
Comment (robertoDiaz) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java]: updated here: https://github.com/liferay-lima/liferay-portal/pull/4839/commits/a074ae1470b618ab8d86d31c6a6a05216a31cc32

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by robertoDiaz** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:stable

**Comment by SylviaLuan** [comment]
Hi @robertoDiaz,

Found a issue that related to this pull request.
[modules-semantic-versioning-jdk8/0/0](https://testray.liferay.com/home/-/testray/case_results/2449206459?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2449197794%26testrayRunId%3D2449203388%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

**Comment by robertoDiaz** [comment]
I'll update ASAP

**Comment by robertoDiaz** [comment]
ci:test:stable

**Comment by SylviaLuan** [comment]
Hi @robertoDiaz,

This [commit](https://github.com/liferay-lima/liferay-portal/pull/4839/commits/fb38e43dbae3bed02a8a83f6f7328628662d03f9) has conflict with the upstream.
```
Auto-merging modules/apps/document-library/document-library-api/bnd.bnd
CONFLICT (content): Merge conflict in modules/apps/document-library/document-library-api/bnd.bnd
Auto-merging modules/apps/document-library/document-library-api/src/main/resources/com/liferay/document/library/configuration/packageinfo
CONFLICT (content): Merge conflict in modules/apps/document-library/document-library-api/src/main/resources/com/liferay/document/library/configuration/packageinfo
error: could not apply fb38e43... LPS-199854 Baseline
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply fb38e43... LPS-199854 Baseline
```

**Comment by robertoDiaz** [comment]
ci:test:stable

**Comment by adolfopa** [comment]
ci:test:stable

**Comment by SylviaLuan** [comment]
ci:forward:force

**Comment by SylviaLuan** [comment]
ci:forward:force

**Pull request closed without merging**